modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/ActionRequestProcessor.java

```diff
@@ -109,20 +108,9 @@ protected void handleRequestInternal(RpcContext rpcCtx, Node node, ActionRequest
                 command = commandsMarshaller.unmarshall(writeRequest.command());
             }
 
-            if (fsm.getListener() instanceof BeforeApplyHandler) {
+            if (listener instanceof BeforeApplyHandler) {
```

Comment:
I will repeat myself. I really don't like an idea of changing raft internals in order to inline some external concepts.